### PR TITLE
Word parsing

### PR DESCRIPTION
**Pull request by DangMin**

### CODE REVIEW

File: web/lib/api/v1/[word].js [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+
+    if (/^@(.*?)(\/.*?\/)(\s+\.*\\s+\/.*\/)?/gms.test(first)) {
+      let parsed = [...first.matchAll(/^@(.*?)(\/.*?\/)(\s+\(.*\)\s+\/.*\/)?/g)]
+      word = parsed[1]
+      ipa = parsed[2]
+      ipaVariations = parsed[3] || null
+    } else {
+      let parsed = [...first.matchAll(/^@(.*?)/gms)]
+      word = parsed[1]
+    }
```
Comment (kcjpop) [inline web/lib/api/v1/[word].js:20]: Okay so there are words that don't have ipa 🤔

Comment (DangMin) [reply web/lib/api/v1/[word].js]: yep, and this is a very small sample

File: web/lib/api/v1/[word].js [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+
+  if (matched !== undefined || matched !== null) {
+    const [first, ...rest] = matched.split('<br>')
+    let word, ipa, ipaVariations
+
+    if (/^@(.*?)(\/.*?\/)(\s+\.*\\s+\/.*\/)?/gms.test(first)) {
+      let parsed = [...first.matchAll(/^@(.*?)(\/.*?\/)(\s+\(.*\)\s+\/.*\/)?/g)]
+      word = parsed[1]
+      ipa = parsed[2]
+      ipaVariations = parsed[3] || null
```
[the remark this answers is not in the document]
Comment (DangMin) [reply web/lib/api/v1/[word].js]: `armor /${ipa}/ - alternative armour /${ipaVariation}`

**Comment by kcjpop** [comment]
I'd prefer this PR to do parsing only, not implementing the API endpoint. That should go to #3. Also we already have `yarn.lock` so `web/package-lock.json` is redundant.

I will take a closer look to the parsing implementation later 😃

**Comment by DangMin** [comment]
Will add package-lock.json into .gitignore. API Endpoints is for testing and presentation purpose only, can be removed or moved to #3 later.

**Comment by kcjpop** [comment]
Adding `package-lock.json` isn't the solution here as it's a lockfile and it should be versioned 😃 The point here is the project is already set up with Yarn so adding a file from npm is a no-no.
+1 for removing API endpoint in this PR. It's better to keep the PR focusing in one thing only, i.e. parsing.

**Review by kcjpop — APPROVED** [review]
Không hiểu gì hết nhưng approve luôn, let's see how things go in prod 😃

**Comment by DangMin** [comment]
Too many conflict between branches `main`, `dev`, `word-parsing`

**Pull request closed without merging**